_posts/2015-12-01-Property-based-testing-FsCheck-Resharper-NCrunch-Visual-Studio-2013.md

```diff
@@ -0,0 +1,159 @@
+    src: "/assets/img/custom/blog/2015-12-01-Property-based-testing/2015-12-01-Faran-crane-truck.jpg"
+    attribution:
+        text: Faran crane truck (13) by Aleksander Stein
+        href: https://www.flickr.com/photos/aleksander_se/4395710462/in/photolist-7Grazq-u8FeRe-aqN7wn-quFTva-59p1jo-fCcGpb-7pZsgL-wQaXfD-8dcvpw-7GravJ-55WvHY-mQNY6-cavBq7-tatNgX-ybwDP-73JJXP-zbo2F9-x74s1Q-5RuhNR-7K2Ppz-59KLFw-6RoBFb-hJCjcV-qMhRKv-oD2dvE-b4TNGV-56ah5u-6bGXrN-e9qotr-4Bz52Z-7K6KNU-ma811D-bBeGjZ-rbmUEf-4TsmqM-ejPJ3o-3YCK6v-55WzgW-iKD8Yb-dCt5Er-55WA8A-cavy1G-8dGnzC-55SmLZ-4bSdwh-9WUchN-4Bz4PF-6UVh7x-cavM6y-6h2Q2n/
+tags:
+- craftsmanship
+- functional programming
+- dotnet
+- fsharp
+- fscheck
+- property based testing
+- visual studio
+---
+
+# Property based testing with FsCheck, Resharper and NCrunch in Visual Studio 2013
+
+### Background
+Craftsmanship in Software strives for high, predictable, repeatable and affordable quality; this is the reason why we care about design, code reviews and of course testing. 
+Test Driven Design helps us create a flexible implementation that meets the expectations of our clients in terms of features, performance and quality; as professionals we have to master as many tools as possible to reach our goal of steadily providing value. 
+	
+### Property based testing
+The above is the reason why I am exploring Property Based Testing; I like to think about it as *"Let the machine do the heavy lifting for you"* or as [John-Huges] says *"Don't write tests, generate them"*.
+
+I am not claiming that you should stop writing unit tests, I find that when driving the design Outside-In hence declaring the relations between my components, unit tests are the tool to use.
+When I want to discover deeper properties of the domain as well as address edge cases that no human can think of, property based testing could be a viable solution.
+	
+If you are asking "How can I set up a development environment in .Net that would allow me to experiment and work with this tool?" the following tutorial is for you. 
+
+### Development environment details
+
+* Windows 10 Home Edition
+* Visual Studio 2013 version 12 Update 5
+* Resharper version 8.2.3
+* NCrunch version 1.48.0.5
+* FsCheck version 2.2.2
+* FsCheck.NUnit version 2.2.2
+* NUnit version 2.6.4
+
+### Steps
+1. Open Visual Studio and create an F# Project
+2. Open the Nuget Package Manager following *TOOLS > Nuget Package Manager > Package Manager Console* from the main menu and type
+    1. ```Install-Package FsCheck```
+    2. ```Install-Package FsCheck.Nunit```
+3. Comment out the content in **FsCheckAddin.fs** file to allow NCrunch to run the tests
+    ``` fs
+        //[<NUnitAddin(Description = "FsCheck addin")>] 
+		//type FsCheckAddin() = 
+		//  interface IAddin with 
+		//      override x.Install host = 
+		//          let tcBuilder = new FsCheckTestCaseBuilder() 
+		//          host.GetExtensionPoint("TestCaseBuilders").Install(tcBuilder) 
+        //          true 
+   ```
```

Comment:
The problem was that I needed a blank line before the ```
Try refreshing plz
I am still working on the links

2015-11-25 17:45 GMT+00:00 Sandro Mancuso <notifications@github.com>:

> In
> _posts/2015-12-01-Property-based-testing-FsCheck-Resharper-NCrunch-Visual-Studio-2013.md
> <https://github.com/codurance/site/pull/78#discussion_r45897308>:
>
> > +
> > +### Steps
> > +1. Open Visual Studio and create an F# Project
> > +2. Open the Nuget Package Manager following *TOOLS > Nuget Package Manager > Package Manager Console* from the main menu and type
> > +    1. ```Install-Package FsCheck```
> > +    2. ```Install-Package FsCheck.Nunit```
> > +3. Comment out the content in **FsCheckAddin.fs** file to allow NCrunch to run the tests
> > +    ``` fs
> > +        //[<NUnitAddin(Description = "FsCheck addin")>]
> > +		//type FsCheckAddin() =
> > +		//  interface IAddin with
> > +		//      override x.Install host =
> > +		//          let tcBuilder = new FsCheckTestCaseBuilder()
> > +		//          host.GetExtensionPoint("TestCaseBuilders").Install(tcBuilder)
> > +        //          true
> > +   ```
>
> Content above showing on a single line. Try removing space between ``` and
> fs
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/codurance/site/pull/78/files#r45897308>.
>



-- 

alessandro di gioia

___________________________________________________

Il presente messaggio di posta elettronica è ad esclusivo utilizzo del
destinatario indicato in indirizzo e gli eventuali documenti allegati
potrebbero avere carattere riservato. Qualora non foste il destinatario del
presente messaggio Vi preghiamo di non leggerlo, di cancellarlo dal Vostro
sistema assieme ad ogni documento ad esso allegato e di volerci avvertire
immediatamente tramite posta elettronica o telefonicamente.

This e-mail is for the sole use of the intended recipient and any file
transmitted with it may contain material that is confidential and
privileged. If you are not the intended recipient of this e-mail, please do
not read this e-mail and delete this message and any file attached from
your system and then notify us immediately by reply e-mail or by telephone.